### ISSUE TITLE
Emulator: when deep link to LUIS hangs due to missing info, no message appears

### ISSUE THREAD

**Issue by mewa1024** [Microsoft/BotFramework-Emulator#948]
## Version
Mac: Emulator_OSX_v4.0.0-preview.40025_40380

## Describe the bug
Deep links to LUIS can fail if no version number is given. When this happens, there is no feedback in the Emulator that the deep link is failing or why. Although there is validation in Emulator that a version is supplied, if you create/edit the .bot file elsewhere, it's possible that the LUIS version might be missing.

Related: As of 9/26, the Basic Bot template in Ibiza did not contain a version for LUIS. Filed a bug on  Ibiza: https://fuselabs.visualstudio.com/BotFramework/_queries/edit/25005/?triage=true

## To Reproduce
1. Open a .bot file that has a LUIS service but no version number (see above about Basic Bot template)
2. Open the resources pane and double-click the LUIS app.
2. Browser opens.
3. If you are logged in to LUIS, the "Loading page ..." toast appears and never goes away.

## Expected behavior
Get a modal that says something like,
**Unable to connect to LUIS app**
Your LUIS configuation is missing a version number. Please edit your LUIS configuration"

## Screenshots
Clicking on the LUIS app in the service pane:
![screen shot 2018-10-01 at 1 56 33 pm](https://user-images.githubusercontent.com/39317542/46315825-5018f280-c583-11e8-9235-b502495b425a.png)
Progress bar in LUIS:
![screen shot 2018-10-01 at 1 56 48 pm](https://user-images.githubusercontent.com/39317542/46315933-91a99d80-c583-11e8-8dbd-811c486b7665.png)
LUIS configuration modal showing that the version was never specified:
![screen shot 2018-10-01 at 1 56 25 pm](https://user-images.githubusercontent.com/39317542/46315963-a8e88b00-c583-11e8-810b-86c8a5d390c4.png)

[bug]

**Comment by justinwilaby** [comment]
Is this a request somehow check the result of opening an external link to a web page opened in another process, e.g. a separate chrome or Edge browser? If so, this isn't necessarily possible out of the box since after the handoff, the Emulator will not have access to the external browser process, nor to the contents of a web page opened in it. If this is important to implement, we can hit the LUIS api with the supplied LUIS service version and display an error message if the API sends back a 404. Of course this means there will be a small delay after double clicking on the service and it opening.

**Comment by mewa1024** [comment]
Hmm. I don’t quite understand the suggestion below, but the problem from the user POV is that deep links to the LUIS model created by the Basic Bot template in Ibiza fail without explanation (or rather, in LUIS.ai users will see a progress bar that will never ever ever ever ever end).

Could we do a simple solution for now? something like

  1.  User clicks the deep link to LUIS from Services list
  2.  Emulator checks if the LUIS version is given.
  3.  If the version number is missing, show a warning modal to add the version number.
  4.  When the user dismisses the warning modal, the LUIS config window opens with the version number shown as missing (validation error below the empty text field.)

My concern is the Basic Bot template in Ibiza (as of last week), creates a .bot file that is missing the LUIS version, so anyone who created a Basic Bot last week will not be able to deeplink from the Emulator, and they will not know why. I’ve filed a bug on Ibiza to fix the template, but it would be good to cover the Emulator end of it as well.

Melanie

**Comment by justinwilaby** [comment]
Holding off on this task until we have a proper service validation strategy for all linkable services.

**Comment by cwhitten** [comment]
@mewa1024 I'd like to just validate the fields in the connection form that have an asterisk next to them. OK?

**Comment by ivanov-dan** [comment]
@mewa1024 Can you still repro this issue?

**Comment by mewa1024** [comment]
@ivanov-dan yes, I was able to repro. I opened a bot that has a .bot file with a LUIS app without a version. From Emulator, when I double-clicked the LUIS app in the sidebar, the deeplink opened to this page and then hung:
![image](https://user-images.githubusercontent.com/39317542/57908431-cb8a9400-7834-11e9-90db-db83df71a220.png)

Here's the .bot file, LUIS section highlighted, no version specified:
![image](https://user-images.githubusercontent.com/39317542/57908536-0f7d9900-7835-11e9-8b18-fda8af0b8305.png)

**Comment by mewa1024** [comment]
@cwhitten Not sure I totally understand the question about the Fabric message bar. You mean can we use the message bar to tell the user that the version is missing? Can we use the Notifications pane instead? Also, can we not open the browser at all if we know a field is missing from the .bot file? The way it is now, when the browser opens, I'm no longer looking at Emulator and might not see the error message anyway.
